llvm/lib/Target/AIE/AIEHazardRecognizer.cpp

```diff
@@ -441,6 +442,19 @@ ScheduleHazardRecognizer::HazardType AIEHazardRecognizer::getHazardType(
       FUDepthLimit));
 }
 
+bool AIEHazardRecognizer::checkConflict(
+    const ResourceScoreboard<FuncUnitWrapper> &Scoreboard, MachineInstr &MI,
+    int DeltaCycles) const {
+  const MCInstrDesc &Desc = MI.getDesc();
+  const unsigned SchedClass =
+      TII->getSchedClass(Desc, MI.operands(), MI.getMF()->getRegInfo());
+  const MemoryBankBits MemoryBanks = getMemoryBanks(&MI);
+  return checkConflict(
+      Scoreboard, ItinData, SchedClass,
+      getSlotSet(Desc, *TII->getFormatInterface(), IgnoreUnknownSlotSets),
+      MemoryBanks, TII->getMemoryCycles(SchedClass), DeltaCycles, std::nullopt);
+}
```

Comment:
I'm probably being pedantic, but do we really need one more wrapper if it's called a single time within the new pipeliner code?

Reply:
We could talk about this. I would vouch for dropping the less generic wrappers. I guess that public interfaces using Descr could be dropped in favour of those using MachineInstr. Descr used to be sufficient, but since we use operandinfo to compute memory banks and iteneraries, MachineInstr is more appropriate.

Reply:
We'll fix interfaces in a follow-up. Closing this discussion